lib/util/tokenUtils.js

```diff
@@ -18,6 +18,51 @@ function expectToken(expectedStart, parseContext)
   }
 }
 
+/**
+ * Raises an error if the start token does not match any of the expected tokens.
+ */
+function expectAnyToken(expectedTokens, parseContext)
+{
+  var tokens = parseContext.tokens;
+
+  var foundToken = false;
+  var firstToken = tokens[0];
+  for (let i = 0; i < expectedTokens.length; i ++)
+  {
+    if(firstToken === expectedTokens[i])
+    {
+      foundToken = true;
+      break;
+    }
+  }
+
+  if(!foundToken)
+  {
+    throw new Error(`Invalid parse state! Expected any of: '${expectedTokens}' but got: '${tokens[0]}' from chain: [${tokens}]. ${parseInfo(parseContext)}`);
+  }
+}
+
+/**
+ * Joins the set of tokens into a string such that joinTokens([a, b]) returns "a b".
+ * Consumes all tokens from the given array such that the array will be empty if it is referenced after calling this function.
+ */
+function joinTokens(tokens)
+{
+  var tokenString = '';
+
+  let token;
+  while(token = tokens.shift())
+  {
+    tokenString += token;
+    if(tokens.length > 0)
+    {
+      tokenString += ' ';
+    }
+  }
+
+  return tokenString;
+}
+
 module.exports = {
-    expectToken, parseInfo
+    expectToken, parseInfo, expectToken, expectAnyToken, joinTokens
```

Comment:
expectToken is in here twice?

Reply:
We depend on it enough, figured having a duplicate export justified it.

Jk, copy pasta. I’ll remove!